clients/src/main/java/org/apache/kafka/common/record/ByteBufferLogInputStream.java

```diff
@@ -38,80 +36,34 @@
         this.maxMessageSize = maxMessageSize;
     }
 
-    public ByteBufferLogEntry nextEntry() throws IOException {
+    public MutableRecordBatch nextBatch() throws IOException {
         int remaining = buffer.remaining();
         if (remaining < LOG_OVERHEAD)
             return null;
 
         int recordSize = buffer.getInt(buffer.position() + Records.SIZE_OFFSET);
-        if (recordSize < Record.RECORD_OVERHEAD_V0)
-            throw new CorruptRecordException(String.format("Record size is less than the minimum record overhead (%d)", Record.RECORD_OVERHEAD_V0));
+        if (recordSize < LegacyRecord.RECORD_OVERHEAD_V0)
+            throw new CorruptRecordException(String.format("Record size is less than the minimum record overhead (%d)", LegacyRecord.RECORD_OVERHEAD_V0));
         if (recordSize > maxMessageSize)
             throw new CorruptRecordException(String.format("Record size exceeds the largest allowable message size (%d).", maxMessageSize));
 
         int entrySize = recordSize + LOG_OVERHEAD;
         if (remaining < entrySize)
             return null;
 
+        byte magic = buffer.get(buffer.position() + LOG_OVERHEAD + LegacyRecord.MAGIC_OFFSET);
```

Comment:
Maybe we should have a shared constant for the magic offset since it's the same for all formats?